frontend/src/App.js

```diff
@@ -1,20 +1,24 @@
 import React from 'react';
 import './App.css';
-import { BrowserRouter as Router, Switch, Route} from 'react-router-dom';
+import { Router, Switch, Route } from 'react-router-dom';
+import MapViewPage from './pages/MapViewPage';
+import { createBrowserHistory } from 'history';
 import Welcome from './components/Welcome';
 
 
+const history = createBrowserHistory();
+
 function App() {
   return (
-    <Router>
+    <Router history={history}>
       <div>
         <Switch>
           <Route exact path="/"  component={Welcome} />
-          {/* <Route exact path="/profile" component={Profile} /> */}
+          <Route path="/map" component={MapViewPage} />
```

Comment:
I think we don't need this route because in order to view the map, the user has to go through welcome page. If i use the URL /map I'm not asked to enter university. So i only get a blank page. You can remove it and the app will still function as intended. If this doesn't make sense, ping me.